#! /usr/bin/env python
# configuration properies of the vrmagic camera

PACKAGE='vrmagic_multi_driver'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#	    Name                    Type      Reconfiguration level             Description                                                                                      Default    Min   Max
gen.add("fps", double_t, SensorLevels.RECONFIGURE_RUNNING, "desired frames per second",0.5, 0.1, 10.0)
gen.add("exposureTime", double_t, SensorLevels.RECONFIGURE_STOP, "exposure time", 30.0, 0.0, 100.0)
gen.add("gainLeft", int_t, SensorLevels.RECONFIGURE_RUNNING, "gain on left sensor", 0, 16, 64)
gen.add("gainRight", int_t, SensorLevels.RECONFIGURE_RUNNING, "gain on right sensor", 0, 16, 64)

gen.add("useLEDs", bool_t, SensorLevels.RECONFIGURE_RUNNING, "turn on/off status leds", True)


exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "CamParams"))

